s3_storage_provider.py

```diff
@@ -84,6 +84,19 @@ def __init__(self, hs, config):
         if "session_token" in config:
             self.api_kwargs["aws_session_token"] = config["session_token"]
 
+        if "request_checksum_calculation" in config or "response_checksum_validation" in config:
+            from botocore.config import Config
+            request_checksum_calculation = None
+            response_checksum_validation = None
+            if "request_checksum_calculation" in config:
+                request_checksum_calculation = config["request_checksum_calculation"]
+            if "response_checksum_validation" in config:
+                response_checksum_validation = config["response_checksum_validation"]
+            self.api_kwargs["config"] = Config(
+                response_checksum_validation=response_checksum_validation,                
+                request_checksum_calculation=request_checksum_calculation
+            )
```

Comment:
nit: This can be written more simply as:

```suggestion
            self.api_kwargs["config"] = Config(
                response_checksum_validation=config.get("response_checksum_validation"),                
                request_checksum_calculation=config.get("request_checksum_calculation")
            )
```